WordPress/gutenberg
# Patterns: Add user categories to site editor sidebar navigation screen

## What?
Lists user patterns by category in the site editor patterns UI.

Depends on [User Patterns Category Select PR](https://github.com/WordPress/gutenberg/pull/53835).

## Why?
With the [addition the ability for users to categorise their patterns](https://github.com/WordPress/gutenberg/pull/53835) we also need a way for them to be displayed by category in the site editor patterns page.

## How?
Adds the user added categories to the category listing in left sidebar of site editor patterns page.

This PR was part of a group of PRs that were merged back into https://github.com/WordPress/gutenberg/pull/53835 before being merged into trunk.

A full list of the PRs involved in this work can be [found here](https://github.com/WordPress/gutenberg/issues/53164).

## Testing Instructions
 - Add several user patterns, synced and unsynced, with categories assigned (make sure to hit enter or add ',' after each category - this is the default tag input component behaviour - we will fix this in a follow up PR)
 - Go to the site editor patterns page and make sure the categories appear in the left nav panel
 - Make sure that when the categories are selected the correct patterns display
 - Make sure the theme patterns also still display as expected
 - Make sure that all the patterns display under 'All patterns' and that searching works as expected
 - Duplicate a theme pattern and make sure the new pattern is in the same category
 - Add a user pattern with category of `Footer` and make sure it displays along with the theme Footer patterns (be aware that although the TT3 theme footer patterns are listed under `Footers` the category slug is actually `footer` so the user category name needs to match this).


## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/3629020/736bf50d-f8d0-4a31-91f8-31ecb132d4b3
